compiler/next/include/chpl/parsing/parsing-queries.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_PARSING_PARSING_QUERIES_H
+#define CHPL_PARSING_PARSING_QUERIES_H
+
+#include "chpl/queries/Context.h"
+#include "chpl/queries/ID.h"
+#include "chpl/queries/Location.h"
+#include "chpl/uast/ASTNode.h"
+#include "chpl/uast/Builder.h"
+#include "chpl/uast/Expression.h"
+#include "chpl/uast/Module.h"
+
+#include <vector>
+
+namespace chpl {
+namespace parsing {
+
+
+  struct FileContents {
+    std::string text;
+    ErrorMessage error;
+    FileContents()
+      : text(), error() { }
+    FileContents(std::string text)
+      : text(std::move(text)), error() { }
+    FileContents(std::string text, ErrorMessage error)
+      : text(std::move(text)), error(std::move(error)) { }
+  };
+  const FileContents& fileText(Context* context, UniqueString path);
```

Comment:
Yes - it is a query managed by the query framework and those can't be constructors (because the query framework needs to be able to call them when recomputing a dependency).